### PR TITLE
Add /update_project endpoint

### PR DESCRIPTION
**Pull request by elsiel23**
With PUT semantics - clears all existing constraints and priorities and sets new values provided by the front end.

### CODE REVIEW

**Comment by elsiel23** [comment]
@thequeenofspades Is this the update behavior you're imagining for the front end? I decided to clear all the existing field priorities and constraint values in this version, but we can change that

**Review by riecke — COMMENTED** [review]
I think we should be more clear about how we use the REST framework.

https://www.django-rest-framework.org/api-guide/requests/#method
https://www.arubanetworks.com/techdocs/AOS-CX/10.10/HTML/rest_v10-0x/Content/Chp_REST_ref/wri-met-pos-put-del-10.htm#:~:text=POST%20creates%20a%20resource.,DELETE%20removes%20a%20resource.

The update method here has PUT semantics (unset fields that are not present), as opposed to PATCH (don't unset fields that are not present). 

If you decide this should have PUT semantics, you should check "request.method" to make sure the frontend wants that semantics.  We can add PATCH later on, since that's an optimization (the client doesn't have to GET first).

**Comment by thequeenofspades** [comment]
Thanks Jon for the detailed explanation! I think in the long run we'd prefer PATCH semantics, since we're autosaving quite often and it's less data to send over the wire. But PUT semantics are fine for a first pass.

File: src/planscape/plan/views.py [hunk 1]
```diff
@@ -240,34 +241,39 @@ def create_project(request: HttpRequest) -> HttpResponse:
...
+            project = Project.objects.get(id=project_id)
+            if project.owner != owner:
+                raise ValueError(
+                    "You do not have permission to view this project.")
+
+            project.priorities.clear()
+            _save_project_parameters(body, project)
+            project.save()
+            return HttpResponse(str(project.pk))
+        else:
```
Comment (riecke) [inline src/planscape/plan/views.py:274]: Nit: the "else" is unnecessary as long as you unindent the next line.

Consider doing the error check first:

  if request.method != "PUT":
      raise KeyError(
                "HTTP methods other than PUT are not yet implemented")

Then you won't need as much indentation.
  if

**Review by riecke — APPROVED** [review]

**Review by thequeenofspades — APPROVED** [review]

File: src/planscape/plan/views.py (hunk 1 above)
Comment (elsiel23) [reply src/planscape/plan/views.py]: done!

**Pull request merged**